llvm/lib/MC/MCParser/AsmParser.cpp

```diff
@@ -3125,6 +3125,16 @@ bool AsmParser::parseDirectiveAscii(StringRef IDVal, bool ZeroTerminated) {
     do {
       if (parseEscapedString(Data))
         return true;
+
+      // Warn about newline characters in parsed string like GAS.
+      size_t NewlinePos = -1;
+      size_t DataSize = Data.size();
+      const char *Start = getTok().getLoc().getPointer() - DataSize - 1;
```

Comment:
The location will be incorrect if the string contains escaped characters (e.g. '\xFF'), and I don't see a way how you can reliably get the location of a newline without making `parseEscapedString` return it. Moreover, the string could contain '\n', which must not be diagnosed.

Reply:
I think it may make sense to look for newlines and emit warnings inside `parseEscapedString`. If not all callers of the function want these warnings, they can be issues conditionally based on a new boolean argument, but I couldn't find such a caller.

Reply:
Moved the code to `parseEscapedString`. Seems that all cases are handled correctly, i've tried to write tests to each of them.